### ISSUE TITLE
getargspec was finally removed (will not be in 3.6) so time to RF "with ignore_warnings()" for it

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#391]
*Originally reported by* **Yaroslav Halchenko (Bitbucket: [yarikoptic](https://bitbucket.org/yarikoptic), GitHub: [yarikoptic](https://github.com/yarikoptic))**

----------------------------------------

I see it still used in the current master:

```
$> git grep -2 getargspec 
igor.py-    """
igor.py-    with ignore_warnings():
igor.py:        argspec = inspect.getargspec(function)
igor.py-    return bool(argspec[1]), len(argspec[0])
igor.py-

```
which leads to fails on travis with "daily" build of python, e.g. https://travis-ci.org/datalad/datalad/jobs/73429087#L816

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/391

**Comment by nedbat** [comment]
Thanks.  What is "RF"? And do you have a replacement for getargspec?

**Comment by nedbat** [comment]
*Original comment by* **Yaroslav Halchenko (Bitbucket: [yarikoptic](https://bitbucket.org/yarikoptic), GitHub: [yarikoptic](https://github.com/yarikoptic))**

----------------------------------------

ah, sorry for being cryptic RF == refactor

replacement -- didn't check in detail but it should be inspect.signature  if I got it right (was glad to see that failure wasn't due to our code ;) )

**Comment by nedbat** [comment]
Fixed in [93c34908d4f5659f3b37ea830536cb5a78086ced (bb)](https://bitbucket.org/ned/coveragepy/commits/93c34908d4f5659f3b37ea830536cb5a78086ced), thanks to Yury Selivanov

**Comment by nedbat** [comment]
Issue #401 was marked as a duplicate of this issue.
